theano/scan_module/tests/test_scan.py

```diff
@@ -625,12 +625,11 @@ def f_rnn(u_t, x_tm1, W_in, W):
         utt.assert_allclose(theano_out1, v_out1)
         utt.assert_allclose(theano_out2, v_out2)
 
-        topo = f2.maker.fgraph.toposort()
-        scan_node = [node for node in topo
-                     if isinstance(node.op, theano.scan_module.scan_op.Scan)]
-        assert len(scan_node) == 1
-        scan_node = scan_node[0]
-        assert scan_node.op.gpu
+        # Ensure f2 contains only one scan node and it's been moved to the GPU
+        scan_nodes = self.scan_nodes_from_fct(f2)
+        assert len(scan_nodes) == 1
+        assert (scan_nodes[0].op.info.get('gpu', False) or
+                scan_nodes[0].op.info.get('gpua', False))
```

Comment:
The goal was to make tests able to test both backends but after chatting with @nouiz, I'll proceed differently in a follow-up PR. The tests with the cuda backend will be moved to a different test class with attributes refering to the cuda backend and another test class will inherit from it and redefine the attributes to point to the GpuArray backend. 

I removed the contentious commit from this PR and only kept the one with which you didn't seem to have any issue.